src/System.Console/src/System/ConsolePal.Unix.cs

```diff
@@ -251,25 +254,37 @@ private struct TerminalColorInfo
             /// <summary>The cached instance.</summary>
             public static TerminalColorInfo Instance { get { return _instance.Value; } }
 
-            /// <summary>Lazy initialization of the terminal color information.</summary>
-            private static Lazy<TerminalColorInfo> _instance = new Lazy<TerminalColorInfo>(() =>
+            internal TerminalColorInfo(TermInfo.Database db)
             {
-                TermInfo.Database db = TermInfo.Database.Instance;
-                TerminalColorInfo tci = new TerminalColorInfo();
                 if (db != null)
                 {
-                    tci.ForegroundFormat = db.GetString(TermInfo.Database.SetAnsiForegroundIndex);
-                    tci.BackgroundFormat = db.GetString(TermInfo.Database.SetAnsiBackgroundIndex);
-                    tci.ResetFormat =
+                    ForegroundFormat = db.GetString(TermInfo.Database.SetAnsiForegroundIndex);
+                    BackgroundFormat = db.GetString(TermInfo.Database.SetAnsiBackgroundIndex);
+                    ResetFormat =
                         db.GetString(TermInfo.Database.OrigPairsIndex) ??
                         db.GetString(TermInfo.Database.OrigColorsIndex);
 
                     int maxColors = db.GetNumber(TermInfo.Database.MaxColorsIndex);
-                    tci.MaxColors = // normalize to either the full range of all ANSI colors, just the dark ones, or none
+                    MaxColors = // normalize to either the full range of all ANSI colors, just the dark ones, or none
                         maxColors >= 16 ? 16 :
                         maxColors >= 8 ? 8 :
                         0;
                 }
+                else
+                {
+                    Debug.Assert(false);
```

Comment:
That said, we do we need this else at all?  Can we just assert at the beginning of the ctor that db is not null?